### PR TITLE
adding keep_count column to library table

### PR DESCRIPTION
**Pull request by ahsu1230**
@andrewconner @yingjieMiao @leogrim @stkem 

doing the `keep_count` stuff in a couple of steps:
This PR: add `num_keep` column, admin endpoint for filling keep counts to libraries, and adding `keepCount` logic to keeping & unkeeping
(2) after applying evolutions, deploy and call admin endpoint. After that, add logic to `LibraryChecker` to make sure keep counts are consistent
(3) Can start using library.keepCount in certain places (instead of looking through DB).

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala [hunk 1]
```diff
@@ -255,6 +255,8 @@ class KeepInterner @Inject() (
       keepToCollectionRepo.getCollectionsForKeep(internedKeep.id.get) foreach { cid => collectionRepo.collectionChanged(cid, inactivateIfEmpty = false) }
     }
 
+    libraryRepo.save(library.copy(keepCount = library.keepCount + 1))
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala]: nvm, I think this should be instead: `keepCount = keepRepo.countByLibrary(libId)`

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala [hunk 2]
```diff
@@ -372,9 +372,11 @@ class KeepsCommander @Inject() (
     searchClient.updateKeepIndex()
   }
 
-  private def setKeepStateWithSession(keep: Keep, state: State[Keep], userId: Id[User])(implicit context: HeimdalContext, session: RWSession): Keep = {
-    val saved = keepRepo.save(keep withState state)
+  private def deactivateKeepWithSession(keep: Keep, userId: Id[User])(implicit context: HeimdalContext, session: RWSession): Keep = {
+    val saved = keepRepo.save(keep withState KeepStates.INACTIVE)
     log.info(s"[unkeep($userId)] deactivated keep=$saved")
+    val library = libraryRepo.get(keep.libraryId.get)
+    libraryRepo.save(library.copy(keepCount = library.keepCount - 1))
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepsCommander.scala]: same as above

File: kifi-backend/modules/shoebox/app/com/keepit/model/KeepRepo.scala [hunk 3]
```diff
@@ -480,10 +480,11 @@ class KeepRepoImpl @Inject() (
   }
   def latestKeptAtByLibraryIds(libraryIds: Set[Id[Library]])(implicit session: RSession): Map[Id[Library], Option[DateTime]] = {
     val keepsGroupedByLibrary = (for (r <- rows if r.libraryId.inSet(libraryIds) && r.state === KeepStates.ACTIVE) yield r).groupBy(_.libraryId)
-    keepsGroupedByLibrary.map { case (libraryId, keeps) => (libraryId, keeps.map(k => k.keptAt).max) }.list
+    val map = keepsGroupedByLibrary.map { case (libraryId, keeps) => (libraryId, keeps.map(k => k.keptAt).max) }.list
       .collect {
         case (Some(libraryId), maxKeptAt) =>
           (libraryId, maxKeptAt)
       }.toMap
+    libraryIds.map { libId => libId -> map.getOrElse(libId, None) }.toMap
```
Comment (ahsu1230) [inline kifi-backend/modules/shoebox/app/com/keepit/model/KeepRepo.scala:488]: @yingjieMiao same idea as `keepRepo.getCountByLibraries` make sure all `libraryIds` map to something

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala [hunk 4]
```diff
@@ -255,6 +255,8 @@ class KeepInterner @Inject() (
       keepToCollectionRepo.getCollectionsForKeep(internedKeep.id.get) foreach { cid => collectionRepo.collectionChanged(cid, inactivateIfEmpty = false) }
     }
 
+    libraryRepo.save(library.copy(keepCount = keepRepo.getCountByLibrary(library.id.get)))
+
     (isNewKeep, wasInactiveKeep, internedKeep)
```
Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepInterner.scala:260]: mind adding a test to verify library count is correct? Want to make sure there is not cache problem here.

**Pull request merged**